manifests/init.pp

```diff
@@ -2,17 +2,124 @@
+#    type: string
+#    Default: directory
+#
+# * `keys_dir_owner`: Ensures that $keys_dir owner is correct.
+#    type: string
+#    Default: root
+#
+# * `keys_dir_group`: Ensures that $keys_dir group is correct.
+#    type: string
+#    Default: root
+#
+# * `keys_dir_mode`: Ensures the $keys_dir mode is -r-x------.
+#    type: integer
+#    Default: 0500
+#
+# * `public_key_path`: Absolute path to the public key.
+#    type: string
+#    Default: /etc/puppet/keys/public_key.pkcs7.pem
+#
+# * `private_key_path`: Absolute path to the private key.
+#    type: string
+#    Default: /etc/puppet/keys/private_key.pkcs7.pem
+#
+# * `public_key_mode`: Ensures public key permission are -rw-r--r--.
+#    type: integer
+#    Default: 0644
+#
+# * `private_key_mode`: Ensures private key permission are -r--------.
+#    type: integer
+#    Default: 0400
+#
+# * `config_dir`: Directory were we want to store eyaml configuration.
+#    type: string
+#    Default: /etc/yaml
+#
+# * `config_dir_ensure`: Ensures $config_dir is a directory.
+#    type: string
+#    Default: directory
+#
+# * `config_dir_owner`: Ensures $config_dir owner is the expected.
+#    type: string
+#    Default: root
+#
+# * `config_dir_group`: Ensures $config_dir group is the expected.
+#    type: string
+#    Default: root
+#
+# * `config_dir_mode`: Ensures $config_dir mode is -rwxr-xr-x.
+#    type: integer
+#    Default: 0755
+#
+# * `config_ensure`: Ensures config file is a file.
+#    type: string
+#    Default: file
+#
+# * `config_path`: hiera-eyaml config file path.
+#    type: string
+#    Default: /etc/eyaml/config.yaml
+#
+# * `config_owner`: Config file owner.
+#    type: string
+#    Default: root
+#
+# * `config_group`: Config file group.
+#    type: string
+#    Default: root
+#
+# * `config_mode`: Config file mode.
+#    type: integer
+#    Default: 0644
+#
+# * `config_options`: eyaml custom configurations like (extension, datadir)
+#    type: hash
+#    Default: { 'pkcs7_public_key'  => '/etc/puppet/keys/public_key.pkcs7.pem',
+#               'pkcs7_private_key' => '/etc/puppet/keys/private_key.pkcs7.pem',
+#             },
+#
+# * `manage_eyaml_config`: Manage eyaml config file.
+#    type: bool
+#    Default: true
+#
+# * `manage_keys`: Manage eyaml keys creation
+#    type: bool
+#    Default: undef
+#
 class eyaml (
   $package_name        = 'hiera-eyaml',
   $package_provider    = 'gem',
   $package_ensure      = 'present',
-  $keys_dir            = "${::settings::confdir}/keys",
+  $keys_dir            = '/etc/puppet/keys',
   $keys_dir_ensure     = 'directory',
-  $keys_dir_owner      = $::settings::user,
-  $keys_dir_group      = $::settings::group,
+  $keys_dir_owner      = root,
+  $keys_dir_group      = root,
   $keys_dir_mode       = '0500',
-  $public_key_path     = '/etc/puppet/keys/public_key.pkcs7.pem',
-  $private_key_path    = '/etc/puppet/keys/private_key.pkcs7.pem',
+  $public_key_path     = "${keys_dir}/public_key.pkcs7.pem",
```

Comment:
keep the hard coding of /etc/puppet/keys and Puppet now has issue with params using a previous param as a variable.